app/ledger.js

```diff
@@ -2093,6 +2093,19 @@ var networkConnected = underscore.debounce(() => {
  * low-level utilities
  */
 
+var muonWriter = (path, payload) => {
+  muon.file.writeImportant(path, JSON.stringify(payload, null, 2), (success) => {
+    if (!success) return console.log('write error: ' + path)
+
+    if ((quitP) && (!getSetting(settings.PAYMENTS_ENABLED)) && (getSetting(settings.SHUTDOWN_CLEAR_HISTORY))) {
```

Comment:
yes, it's weird: this deals with a race condition that @NejcZdovc found during testing.

it's done that way because the file may be in the process of being written when we want to delete it. in which case, if we unlink the file, we might just then go ahead and write it again. hence, always delete in this case...